### PR TITLE
Update readmes

### PR DESCRIPTION
**Pull request by BPScott**
## Description

Writing words and updating examples so people know how to do things.

- Add readme for typescript plugin
- Update headings of usage sections for eslint/stylelint/prettier plugins
  for consistency - name the usage section "Usage"
- Fix examples in eslint/stylelint/prettier plugins - the demo plugins
  were missing a name
- Remove unused import from babel plugin examples
- In build-library plugin link to the technology plugins that it
  composes

### CODE REVIEW

File: packages/loom-plugin-babel/README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ yarn add @shopify/loom-plugin-babel --dev
 Add the plugin to your project and specify any defaults that you wish to set using the `config` option, which is either an object containing all the options described on [Babel's options page](https://babeljs.io/docs/en/options) with the exception of `include` and `exclude`, or a function that accepts the existing babel options and expects the new babel options object to be returned. This example defines a `config` that uses the `@shopify/babel-preset` preset.
 
 ```js
-import {createPackage, Runtime} from '@shopify/loom';
```
Comment (BPScott) [inline packages/loom-plugin-babel/README.md:18]: These imports were never used

File: packages/loom-plugin-eslint/README.md [hunk 2]
```diff
@@ -41,13 +43,15 @@ This plugin adds the following hooks to `LintWorkspaceConfigurationCustomHooks`:
...
 
-  const plugin = createWorkspaceLintPlugin(({hooks}) => {
-    hooks.configure.hook((configure) => {
-      // Modify the maximum number of allowed warnings from the default of 0
-      configure.eslintFlags?.hook((flags) => ({
-        ...flags,
-        maxWarnings: 5,
-      }));
+  function demoPlugin() {
+    return createWorkspaceLintPlugin('Demo', ({hooks}) => {
```
Comment (BPScott) [inline packages/loom-plugin-eslint/README.md:47]: `createWorkspaceLintPlugin` expects a name as the first argument, so the old example was not valid. This mistake was also present in the prettier and stylelint readmes.

Question: Should we demo using `createWorkspaceLintPlugin` for this or should we use the slightly more verbose `createWorkspace`?

Using `createWorkspaceLintPlugin` saves a level of nesting, but do we care about people to making the connection that you can configure many tasks within `createWorkspace` so that could lead to terser code if for instance they wanted to modify the `lint` and `type-check` task?


Basically, should this be:

```js
function demoPlugin() {
  return createWorkspacePlugin('Demo', ({tasks: {lint}}) => {
    lint.hook(({hooks}) => {
      hooks.configure.hook((configure) => {
        // Modify the maximum number of allowed warnings from the default of 0
        configure.eslintFlags?.hook((flags) => ({
          ...flags,
          maxWarnings: 5,
        }));
      });
    });
  });
}
```

File: packages/loom-plugin-eslint/README.md [hunk 3]
```diff
@@ -10,14 +10,16 @@ This package does not make any assumptions about what ESLint ruleset you should
 yarn add @shopify/loom-plugin-eslint --dev
 ```
 
-## `eslint()`
+## Usage
```
Comment (BPScott) [inline packages/loom-plugin-eslint/README.md:13]: All the other plugins have a `Usage` heading so get in line. This change also happens in the prettier/stylelint readmes

File: packages/loom-plugin-eslint/README.md [hunk 4]
```diff
@@ -10,14 +10,16 @@ This package does not make any assumptions about what ESLint ruleset you should
...
 
-The `eslint` function returns a `loom` plugin. This plugin applies to the workspace, not an individual project.
+Add `eslint` to your loom worksace plugins.
```
Comment (BPScott) [inline packages/loom-plugin-eslint/README.md:15]: I'm not sure how to handle teaching the idea that in a many-project repo you'll have one workspace config file (using `createWorkspace()`), and many project config files, but in a single-project repo you have one config files where you configure both workspace-level and project-level plugins (using `createPackage()` / `createWebApp()`).


Is this even the right place to do that? Is it ok to kinda hand-wave it away here with the comment in the code example below?

Suggestions welcome here.

**Review by BPScott — COMMENTED** [review]
I have questions!

File: packages/loom-plugin-eslint/README.md (hunk 2 above)
Comment (BPScott) [reply packages/loom-plugin-eslint/README.md:47]: I've slept on this and I think I like the `createWorkspace` approach for consistency more.

File: packages/loom-plugin-eslint/README.md (hunk 3 above)
Comment (BPScott) [reply packages/loom-plugin-eslint/README.md:13]: I think long-term that's overkill, we're the team writing packages in this repo, once we wrangle everything to have a consistent style (almost there!) it will be easier to say "hey your new thing doesn't follow existing convention" in PR reviews as we can point to any existing README.

I don't think "should have a heading called usage" is a particularly useful test to run
